### PR TITLE
[SDKS-620] GetTreatmentWithConfig and GetTreatmentsWithConfig implementation

### PR DESCRIPTION
**Pull request by mssplitio**
## Changes

- GetTreatmentWithConfig and GetTreatmentsWithConfig implementation

## Tests
Client simulator:

image

### CODE REVIEW

File: Net SDK Unit Tests/Integration Tests/SelfRefreshingSplitClientTests.cs [hunk 1]
```diff
@@ -8,25 +11,12 @@ namespace Splitio_Tests.Integration_Tests
     [TestClass]
     public class SelfRefreshingSplitClientTests
     {
+        [Ignore]
```
Comment (mmelograno) [inline Net SDK Unit Tests/Integration Tests/SelfRefreshingSplitClientTests.cs]: Are you ignoring this tests?

File: NetSDK/Services/Client/Classes/SplitClient.cs [hunk 2]
```diff
@@ -255,19 +223,94 @@ protected TreatmentResult GetTreatment(Key key, ParsedSplit split, Dictionary<st
...
+                    var treatmentResult = DoGetTreatment(key, feature, attributes, true);
+
+                    treatmentsForFeatures.Add(feature, treatmentResult);
+
+                    ImpressionsQueue.Add(BuildImpression(key.matchingKey, feature, treatmentResult.Treatment, start, treatmentResult.ChangeNumber, LabelsEnabled ? treatmentResult.Label : null, key.bucketingKeyHadValue ? key.bucketingKey : null));
+                }
+
+                if (metricsLog != null)
+                {
+                    metricsLog.Time(SdkGetTreatment, clock.ElapsedMilliseconds);
```
Comment (mmelograno) [inline NetSDK/Services/Client/Classes/SplitClient.cs]: Are you logging `sdk.GetTreatment` in metrics for `getTreatment`, `getTreatments`, `getTreatmentWithConfig` and `getTreatmentsWithConfig`. It should be different for each method.

Comment (mssplitio) [reply NetSDK/Services/Client/Classes/SplitClient.cs]: You are right. Good catch Mati. Thanks!!

**Comment by mssplitio** [comment]
@mmelograno Suggestions applied.

**Comment by NicoZelaya** [comment]
@mmelograno give it a review tomorrow, we can iterate the code as many times as needed. Same for net core.

**Pull request merged**